internal/operator/secret.go

```diff
@@ -0,0 +1,46 @@
+package operator
+
+import (
+	"context"
+	"fmt"
+
+	log "github.com/sirupsen/logrus"
+	corev1 "k8s.io/api/core/v1"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	runtimeClient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func (oe operatorClient) CreateSecret(ctx context.Context, name string, content map[string]string, secretType corev1.SecretType, namespace string) (*corev1.Secret, error) {
+	secret := corev1.Secret{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Secret",
+			APIVersion: "v1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+		},
+		StringData: content,
+		Type:       secretType,
+	}
+	err := oe.Client.Create(ctx, &secret, &runtimeClient.CreateOptions{})
```

Comment:
On oe.Client : oe stands for Openshift engine and that comes from preflight. Let's standardize it as operator or operatorClient or anything like oc if you will. BTW there are other parts of the code already merged like that. We need to scan through that in order to make it more intuitive.